### ISSUE TITLE
Feature Creep and Emulating Python2 Behavior

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/hxrsnd#60]
No immediate action needed, but I wanted to start a thread about feature creep.

There's a lot of stuff in here that looks like it was a request to make things more like the old Python 2 environments. A prime example is the move function:

https://github.com/slaclab/HXRSnD/blob/master/hxrsnd/aerotech.py#L193-L262

Compared to the base EpicsMotor move function:

https://github.com/NSLS-II/ophyd/blob/master/ophyd/epics_motor.py#L120-L161

This is probably reasonable now while the hutch straddles the two systems. However, stuff like this isn't maintainable in the long-term. There shouldn't be an option to not return a status object from a move, for example, because this breaks the API. At what point do we go back and cut the feature creep to retain a manageable code base?

The module is slowly starting to look more like a successful attempt to force us to port half of the old features we were trying to deprecate.

**Comment by apra93** [comment]
This is going to sound like a punt, but I think the only way we're going to be able to get away with it is by getting rid of the python 2 interface or teaching them how to use `ophyd` and `bluesky`. Once we do one these two things we can wean them off the features by leaving a warning (like `ophyd` did with `signal_names`) and then eventually do away with the feature. 

I could be wrong though, and starting to just slowly remove features will go fine. I'm planning on testing this by adding a warning to the `__repr__` in an upcoming PR since this is just not the way it should be used.

**Comment by apra93** [comment]
Been thinking about this again for SnD, and one solution I'm starting to gravitate towards is to have `mv` be more than simply an alias for `move`. I'm imagining we implement `move` the way it is supposed to be (for example always returning status objects) but then to add all the interface wishes of the scientists to `mv`. Seems like a good compromise of maintaining the codebase's internal functionality for `ophyd` and `bluesky` while appeasing the scientists.

**Comment by apra93** [comment]
This would also extend to the other interface methods like `mvr`, `umv` and `__call__`, in that they should all be calling `mv` under the hood since they should only be used by the scientists.

**Comment by teddyrendahl** [comment]
You should look at what we did in the new `pcdsdevices`

https://github.com/pcdshub/pcdsdevices/blob/master/pcdsdevices/mv_interface.py

**Comment by apra93** [comment]
Yeah that's what I mean. Have `move` remain pure but put little requests like optionally returning a status in `mv` so we know it's isolated.

**Comment by teddyrendahl** [comment]
👍
